tools/pyi/gen_pyi.py

```diff
@@ -35,46 +37,32 @@
 read gen_pyi for the gory details.
 """
 
-# TODO: remove after migrating entire codegen to the new data model.
-def should_generate_python_binding(declaration):
-    name = declaration['name']
-    for pattern in SKIP_PYTHON_BINDINGS:
-        if re.match('^' + pattern + '$', name):
-            return False
-
-    simple_types = [arg['simple_type'] for arg in declaration['arguments']]
-    signature = '{}({})'.format(name, ', '.join(simple_types))
-    for pattern in SKIP_PYTHON_BINDINGS_SIGNATURES:
-        if pattern == signature:
-            return False
-
-    return True
-
+# TODO: consider waiting to group by base name until we actually need to (after computing type hint signatures, when adding @overload directives)
+def group_by_base_name(python_funcs: Sequence[PythonSignatureNativeFunctionPair]) -> Dict[str, Sequence[PythonSignatureGroup]]:
```

Comment:
@ljk53 don't you have something similar to this for the Python codegen? It seems to me like pyi grouping should be the same as arg parsing grouping.

Reply:
I left a comment further down (in generate_type_hints) - I think we can get rid of this grouping later, since it doesn't look like the pyi codegen does anything meaningful with groups of overloads. It would probably be cleaner to have the codegen operate per `PythonSignatureGroup`.